### ISSUE TITLE
Improve support debugging

### ISSUE THREAD

**Issue by birkjernstrom** [polarsource/polar#876]
Currently debugging support questions is a bit of a pain. Requiring a lot of copy-paste of different IDs & events and searching through one-by-one until you reach the root cause.

Would be massively improved with:

- [x] Integrate Posthog with Sentry to tie exceptions to session events
- [x] Implement Correlation IDs in logging to tie all logging events to one particular request that can more easily be debugged
- More logging. Ongoing effort vs. one-off fix here.

## Funding
* You can sponsor this specific effort via a [Polar.sh](https://polar.sh) pledge below
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/polarsource/polar/issues/876">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/polarsource/polar/issues/876/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/polarsource/polar/issues/876/pledge.svg">
</picture>
</a>

**Comment by frankie567** [comment]
* [ ] Integrate Sentry for Next.js SSR

**Comment by frankie567** [comment]
Solved by the following PR:

* #1079 
* #1075 
* #1079
* #1083

Basically, I tweaked the setup of Sentry and Posthog so they report more data in the events. They're now tied together so errors are reported in Posthog (with a link to Sentry) and Sentry reports show a link back to the affected user in Posthog.

All logs now also show a correlation ID so they can be grouped easily in monitoring. I also took the opportunity to tweak the configuration so third-party loggers (namely `uvicorn`, `sqlalchemy` and `arq`) share the same one and adopt our formatting. Output is much cleaner this way!
